File: golang/2379. 得到 K 个黑块的最少涂色次数.go
func minimumRecolors(blocks string, k int) int {
    i, j := 0, k - 1
    res := len(blocks)
    for j < len(blocks) {
        cnt := 0
        for h := i; h <= j; h++ {
            if blocks[h] == 'B' {
                cnt++
            }
        }
        
        res=min(res, k-cnt)
        i++
        j++
    }
    return res
}
func min(a, b int) int {
    if a < b {
        return a
    }
    return b
}

File: cpp/2379. 得到 K 个黑块的最少涂色次数.cpp
class Solution {
public:
    int minimumRecolors(string blocks, int k) {
        int i=0,j=k-1;
        int ans=blocks.length();
        while(j<=blocks.length()){
            int cnt=0;
            for(int h=i;h<=j;h++){
                if(blocks[h]=='B'){
                    cnt++;
                }
            }
            ans=min(ans,k-cnt);
            i++,j++;
        }
        return ans;
    }
};